File: 프로그래머스/Level1/완주하지 못한 선수.py
# https://programmers.co.kr/learn/courses/30/lessons/42576

def solution(participant, completion):
    dic = dict()
    
    for parti in participant:
        if parti in dic:
            dic[parti] += 1
        else:
            dic[parti] = 1
    for com in completion:
        dic[com] -= 1
    
    for key in dic:
        if dic[key] != 0:
            return key

File: 프로그래머스/Level1/완주하지 못한 선수.js
// https://programmers.co.kr/learn/courses/30/lessons/42576?language=javascript

function solution(participant, completion) {
  const hash = {};
  participant.forEach((k) => {
    if (!hash[k]) hash[k] = 1;
    else hash[k] += 1;
  });
  completion.forEach((c) => (hash[c] -= 1));
  const temp = [];
  for (const h in hash) {
    temp.push([h, hash[h]]);
  }
  temp.sort((a, b) => b[1] - a[1]);
  return temp[0][0];
}
